### PR TITLE
[descriptor] Ensure that there are no duplicated keys

### PR DESCRIPTION
**Pull request by afilini**
### Description

Most of the code assumes that fingerprints are unique and bad things happen if that's not the case. 

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

#### New Features:

* [x] I've added tests for the new feature
* [x] I've added docs for the new feature
* [ ] I've updated `CHANGELOG.md

### CODE REVIEW

File: src/descriptor/mod.rs [hunk 1]
```diff
@@ -783,5 +801,14 @@ mod test {
             result.unwrap_err(),
             DescriptorError::HardenedDerivationXpub
         ));
+
+        let descriptor = "wsh(multi(2,tpubD6NzVbkrYhZ4XHndKkuB8FifXm8r5FQHwrN6oZuWCz13qb93rtgKvD4PQsqC4HP4yhV3tA2fqr2RbY5mNXfM7RxXUoeABoDtsFUq2zJq6YK/0/*,tpubD6NzVbkrYhZ4XHndKkuB8FifXm8r5FQHwrN6oZuWCz13qb93rtgKvD4PQsqC4HP4yhV3tA2fqr2RbY5mNXfM7RxXUoeABoDtsFUq2zJq6YK/1/*))";
+        let result = into_wallet_descriptor_checked(descriptor, &secp, Network::Testnet);
+
+        assert!(result.is_err());
```
Comment (tcharding) [inline src/descriptor/mod.rs:808]: I don't think we need this, the `unwrap_err` panics if its not an error.

**Review by tcharding — COMMENTED** [review]
One small suggestion and then for what its worth, LGTM :)

File: src/descriptor/mod.rs (hunk 1 above)
Comment (notmandatory) [reply src/descriptor/mod.rs:808]: I think on line 810 he's unwrapping it to assert it's the expected `DescriptorError::DuplicatedKeys`

**Comment by notmandatory** [comment]
utACK 36277a7

**Comment by afilini** [comment]
Also worth pointing out: ColdCard has done the same recently, see https://blog.coinkite.com/version-3.2.1-released/.

File: src/descriptor/mod.rs (hunk 1 above)
Comment (afilini) [reply src/descriptor/mod.rs:808]: Yes, there's also the previous assertion to ensure that it's actually an error. If any of those assertions or unwraps cause a panic the test will fail, which is what we want if something weird is going on.

**Pull request merged**